### PR TITLE
feat(cb2-11360): create new automated cherished transfer lambda

### PR DESCRIPTION
**Pull request by naathanbrown**
## Description

A new lambda to manage automated cherished transfers that will come from a queue.

Related issue: [https://dvsa.atlassian.net/browse/CB2-11360](11360)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/handler/mot-update-vrm.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          },
+          parsedRecord.vrm,
+          currentRecord as TechRecordType<'get'>,
+        );
+
+        await updateVehicle(recordsToArchive, recordsToUpdate);
+
+        logger.info(`Updated systemNumber ${currentRecord.systemNumber} with VRM ${parsedRecord.vrm}`);
+
+        recordsToUpdate.forEach((record) => recordsToSend.push({ ...record, userEmail: 'something@goes.here' }));
```
Comment (cb-cs) [inline src/handler/mot-update-vrm.ts:59]: What is the general concencus on the e-mail that we need to be using?

Comment (naathanbrown) [reply src/handler/mot-update-vrm.ts:59]: You know what, I missed changing this to 'CVS Auto CT blah blah' but on further thoughts what should this actually be? Will raise in SU

File: src/services/sqs.ts [hunk 2]
```diff
@@ -14,7 +16,8 @@ export const addToSqs = async (messageBody: SQSRequestBody, queueUrl: string) =>
   };
 
   try {
-    await sqsClient.send(new SendMessageCommand(params));
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    await sqsClient.send(new SendMessageCommand(params) as any);
```
Comment (Daniel-Searle) [inline src/services/sqs.ts:20]: why do you need to cast it as any?

File: src/handler/mot-update-vrm.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import { MotCherishedTransfer } from '../models/motCherishedTransfer';
+import { SearchCriteria } from '../models/search';
+import { SNSMessageBody } from '../models/updateVrm';
+import { processCherishedTransfer } from '../processors/processCherishedTransfer';
+import { searchByCriteria, updateVehicle } from '../services/database';
+import { publish } from '../services/sns';
+import { StatusCode } from '../util/enum';
+import logger from '../util/logger';
+
+export const handler = async (event: SQSEvent) => {
```
[the remark this answers is not in the document]
Comment (naathanbrown) [reply src/handler/mot-update-vrm.ts:14]: the lint command in the repo has no concern?

File: src/services/sqs.ts (hunk 2 above)
Comment (naathanbrown) [reply src/services/sqs.ts:20]: @kcarr-bjss why is this changed like this? I didn't do this bit so

File: src/handler/mot-update-vrm.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Daniel-Searle) [reply src/handler/mot-update-vrm.ts:14]: Oh maybe it's just may web storm config 😆

File: src/services/sqs.ts (hunk 2 above)
Comment (naathanbrown) [reply src/services/sqs.ts]: donezo

**Review by Daniel-Searle — APPROVED** [review]

**Review by LGin-BJSS — APPROVED** [review]

**Pull request merged**